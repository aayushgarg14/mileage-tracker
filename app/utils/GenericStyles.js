import { StyleSheet } from 'react-native';

const StyleConstants = {
  color: {
    $PRIMARY: '#2E2F32',
    $DARK_GREY: '#565656',
    $GREY: '#676D70',
    $DARK: '#202124',
    $GREEN: '#5BC558',
    $RED: '#E94638',
    $BLUE: '#8AB4F8',
  },
  fontSize: {
    $FONT_XXXL: 24,
    $FONT_XXL: 22,
    $FONT_XL: 20,
    $FONT_L: 18,
    $FONT_ML: 16,
    $FONT_M: 14,
    $FONT_SM: 13,
    $FONT_S: 12,
    $FONT_MS: 11,
    $FONT_XS: 10,
    $FONT_XXS: 8,
  },
};

const GenericStyles = StyleSheet.create({
  bold: { fontWeight: '700' },
  mt4: { marginTop: 4 },
  mt8: { marginTop: 8 },
  mt16: { marginTop: 16 },
  mb4: { marginBottom: 4 },
  mb8: { marginBottom: 8 },
  mb16: { marginBottom: 16 },
  ml4: { marginLeft: 4 },
  ml16: { marginLeft: 16 },
  mr4: { marginRight: 4 },
  mr8: { marginRight: 8 },
  mr16: { marginRight: 16 },
  phMain: { paddingHorizontal: 16 },

  fontXS: { fontSize: 10 },
  fontS: { fontSize: 12 },
  fontM: { fontSize: 14 },
  fontL: { fontSize: 16 },

  lightColor: {
    color: StyleConstants.color.$GREY,
  },
  redColor: {
    color: StyleConstants.color.$RED,
  },
  blueColor: {
    color: StyleConstants.color.$BLUE,
  },

  primaryBGColor: {
    backgroundColor: StyleConstants.color.$PRIMARY,
  },
  blueBGColor: {
    backgroundColor: StyleConstants.color.$BLUE,
  },
  darkBGColor: {
    backgroundColor: StyleConstants.color.$DARK,
  },

  blueBorderColor: {
    borderColor: StyleConstants.color.$BLUE,
  },

  // Flex
  f1: { flex: 1 },
  fdr: { flexDirection: 'row' },
  aic: { alignItems: 'center' },
  asc: { alignSelf: 'center' },
  jcs: { justifyContent: 'space-between' },
  fc: { justifyContent: 'center', alignItems: 'center' },

  viewWrap: { width: 0, flexGrow: 1 },
});

const CircleStyles = size => {
  return StyleSheet.create({
    circle: { width: size, height: size, borderRadius: size / 2 },
  });
};

const DimensionsStyles = (height, width) => {
  return StyleSheet.create({
    dimensions: { width, height },
    width: { width },
    height: { height },
  });
};

export { StyleConstants, GenericStyles, CircleStyles, DimensionsStyles };
