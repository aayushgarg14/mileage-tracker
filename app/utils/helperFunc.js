import dayjs from 'dayjs';

var relativeTime = require('dayjs/plugin/relativeTime');
dayjs.extend(relativeTime);

const formatDayjs = (date, format) =>
  dayjs(date).format(format || 'DD MMM YYYY');

const fromNowDayjs = date => dayjs(date).fromNow();

const getLastMonthDate = () => dayjs().subtract(1, 'month').format('MM YYYY');

const formatDate = date => {
  var d = new Date(date),
    month = '' + (d.getMonth() + 1),
    day = '' + d.getDate(),
    year = d.getFullYear();

  if (month.length < 2) {
    month = '0' + month;
  }

  if (day.length < 2) {
    day = '0' + day;
  }

  return [year, month, day].join('-');
};

const formatTime = date => {
  return new Date(date).toLocaleString('en-US', {
    hour: 'numeric',
    minute: 'numeric',
    hour12: true,
  });
};

export { formatDayjs, fromNowDayjs, formatDate, formatTime, getLastMonthDate };
