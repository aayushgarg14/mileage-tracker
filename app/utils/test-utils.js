import React from 'react';
import { render as rntlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import { store as myStore } from '../store';

const mockStore = configureStore();
const store = mockStore(myStore.getState());

function render(ui, options) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rntlRender(ui, { wrapper: Wrapper, ...options });
}

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { render, store };
