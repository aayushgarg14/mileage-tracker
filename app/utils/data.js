import { StyleConstants } from './GenericStyles';

const inputRules = {
  required: {
    value: true,
    message: 'Enter a number',
  },
  pattern: {
    value: /^\d+(\.\d+)?$/,
    message: 'Not a valid number',
  },
};

const home = {
  gas: {
    title: 'Gas',
    icon: 'gas',
    data: [
      {
        id: 1,
        icon: 'opacity',
        type: 'avgCons',
        name: 'Average fuel consumption',
        iconColor: StyleConstants.color.$BLUE,
        value: 0,
        displayValue: '0',
        initials: 'km/l',
        initial: 'post',
      },
      {
        id: 2,
        icon: 'trend',
        type: 'lastCons',
        name: 'Last fuel consumption',
        iconColor: StyleConstants.color.$BLUE,
        value: 0,
        displayValue: '0',
        initials: 'km/l',
        initial: 'post',
      },
      {
        id: 3,
        icon: 'trend',
        type: 'price',
        name: 'Last fuel price',
        iconColor: StyleConstants.color.$BLUE,
        value: 0,
        displayValue: '0',
        initials: 'Rs.',
        initial: 'pre',
      },
      {
        id: 4,
        type: 'lastUpdate',
        name: '',
      },
    ],
  },
  costs: {
    title: 'Costs',
    icon: 'currency',
    data: [
      {
        id: 1,
        title: 'THIS MONTH',
        type: 'curr',
        data: [
          {
            id: 1,
            icon: 'gas',
            iconColor: StyleConstants.color.$BLUE,
            name: 'Gas',
            type: 'gas',
            value: 0,
            displayValue: '0.00',
            initials: 'Rs.',
            initial: 'pre',
          },
          {
            id: 2,
            icon: 'other',
            iconColor: StyleConstants.color.$BLUE,
            name: 'Other costs',
            type: 'other',
            value: 0,
            displayValue: '0.00',
            initials: 'Rs.',
            initial: 'pre',
          },
        ],
      },
      {
        id: 2,
        title: 'PREVIOUS MONTH',
        type: 'prev',
        data: [
          {
            id: 1,
            icon: 'gas',
            iconColor: StyleConstants.color.$BLUE,
            name: 'Gas',
            type: 'gas',
            value: 0,
            displayValue: '0.00',
            initials: 'Rs.',
            initial: 'pre',
          },
          {
            id: 2,
            icon: 'other',
            iconColor: StyleConstants.color.$BLUE,
            name: 'Other costs',
            type: 'other',
            value: 0,
            displayValue: '0.00',
            initials: 'Rs.',
            initial: 'pre',
          },
        ],
      },
    ],
  },
  entries: {
    title: 'Last entries',
    icon: 'timeline',
    data: [],
  },
};

export { inputRules, home };
