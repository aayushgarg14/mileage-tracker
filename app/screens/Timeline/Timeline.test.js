import React from 'react';
import { render, fireEvent } from '../../utils/test-utils';
import TimelineScreen from './index';

jest.mock('@react-navigation/native', () => ({
  useIsFocused: jest.fn(),
}));

it('renders', () => {
  render(<TimelineScreen />);
});

it('navigates to the AddEntryScreen', () => {
  const navigateMock = jest.fn();
  const { toJSON, getByTestId } = render(
    <TimelineScreen navigation={{ navigate: navigateMock }} />,
  );

  fireEvent.press(getByTestId('Timeline.Button'));

  expect(navigateMock).toBeCalledWith('AddEntry');
  expect(toJSON()).toMatchSnapshot();
});
