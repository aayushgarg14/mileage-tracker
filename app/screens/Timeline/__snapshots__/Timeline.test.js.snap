// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigates to the AddEntryScreen 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#2E2F32",
        "height": 40,
      }
    }
  />,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2E2F32",
        "borderBottomWidth": 0,
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 0,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FFF",
                    "fontSize": 20,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
      <Text
        maxFontSizeMultiplier={1}
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 14,
              "letterSpacing": 0,
            },
            Object {
              "fontWeight": "700",
            },
            Object {
              "fontSize": 16,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      />
    </View>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#202124",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#202124",
            "flex": 1,
            "paddingTop": 20,
          },
          Object {
            "paddingHorizontal": 16,
          },
        ]
      }
    >
      <RCTScrollView
        ListEmptyComponent={[Function]}
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      >
        <View>
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View
              style={
                Object {
                  "marginBottom": 16,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#676D70",
                        "fontSize": 36,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <Text
            maxFontSizeMultiplier={1}
            style={
              Array [
                Object {
                  "color": "#FFF",
                  "fontSize": 14,
                  "letterSpacing": 0,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "color": "#676D70",
                },
                undefined,
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            No data available
          </Text>
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 28,
          "bottom": 16,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "right": 16,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 3,
            },
            Object {
              "borderRadius": 28,
              "elevation": 4,
            },
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="Timeline.Button"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#8AB4F8",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 56,
                "justifyContent": "center",
                "padding": 16,
                "width": 56,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 24,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
